acvm-repo/acir/src/circuit/opcodes.rs

```diff
@@ -146,16 +146,105 @@ pub enum Opcode<F: AcirField> {
 impl<F: AcirField> std::fmt::Display for Opcode<F> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
-            Opcode::AssertZero(expr) => expr.fmt(f),
+            Opcode::AssertZero(expr) => {
+                write!(f, "EXPR ")?;
```

Comment:
So it does look like we will lose a little modularity with getting to share the `Expression` display, but that is alright imo. It would be nice to display all expressions using equations as well. I feel they could share parsing pretty easily, but their printers would definitely differ as we do not always want the simplification. Either way it would be better to follow-up with a general `Expression` display rather than put it in this PR.

Reply:
Ah, I thought we didn't want to show all expressions as equations, for example in an input/output of a memory op. For example this:

```
MEM (id: 0, read at: EXPR [ (1, w0) ], value: EXPR [ (1, w1) ])
```

would show up as:

```
MEM (id: 0, read at: EXPR w0 = 0, value: EXPR w1 = 0
```

(I guess we could include brackets in this case, but `w0 = 0` and `w1 = 0` read a bit weird)

Reply:
I was thinking more like:
```
MEM (id: 0, read at: EXPR [ (1, w0) ], value: EXPR [ (1, w1) ])
```
would show up as:
```
MEM (id: 0, read at: EXPR w0, value: EXPR w1
```